### PR TITLE
README file for`membraneRoof` case 

### PR DESCRIPTION
**Pull request by iBatistic**
## Pull Request Description 🚀

This PR cleans up the `membraneRoof` tutorial, improves the case documentation, and adds a regression test for the partitioned FSI setup.

  Main changes

  - 0.orig-based layout 
  - update Allrun/Allclean and related case files 
  - Added README.md 
  - added src library for inlet boundary condition
  - added regressionTest.sh
    - use a shortened regression run with endTime = 2
    - check the final roof-centre Dy from:
    postProcessing/0/solidPointDisplacement_pointDisp.dat
    - check the final membrane Fy from:
    postProcessing/fluid/forces/0/force.dat

  Note

  The new regression test currently targets the partitioned setup only.

### CODE REVIEW

File: tutorials/fluidSolidInteraction/membraneRoof/system/fluid/refineMeshDict [hunk 1]
```diff
@@ -1,11 +1,9 @@
 /*--------------------------------*- C++ -*----------------------------------*\
-| solids4foam: solid mechanics and fluid-solid interaction simulations        |
```
Comment (philipcardiff) [inline tutorials/fluidSolidInteraction/membraneRoof/system/fluid/refineMeshDict:2]: You will need to run the case header updating script => I just made these changes to the base class, which is why it is flagged here.

You may also need to run the whitespace clean up command too

Comment (iBatistic) [reply tutorials/fluidSolidInteraction/membraneRoof/system/fluid/refineMeshDict:2]: case headers and copyright is updated with scripts.

> You may also need to run the whitespace clean up command too

What command you use for that? I put in emacs .config to clean them when saving the file

Comment (philipcardiff) [reply tutorials/fluidSolidInteraction/membraneRoof/system/fluid/refineMeshDict:2]: You can run this command from the parent directory:
```
git grep --cached -I -l -z '[[:blank:]]\+$' | xargs -0 -r sed -i -e 's/[[:space:]]\+$//'
```

The GitHub action here https://github.com/solids4foam/solids4foam/blob/master/.github/workflows/removeTrailingWhitespace.yml prints out this command if a file fails; maybe we should just get it to fix it rather than print an error. Actually, it would probably be useful if this command and update-header/copyright commands were also automatically run => I am sure codex can help

Comment (iBatistic) [reply tutorials/fluidSolidInteraction/membraneRoof/system/fluid/refineMeshDict:2]: Ah yes, i forgot we have that Github action. I ran the command and it didn’t report any trailing whitespace.
I’ll go ahead and mark this conversation as resolved.

**Comment by iBatistic** [comment]
@philipcardiff  the build tests are effectively compiling the`feature-petsc-snes-quasi-monolithic` branch, which is still being actively developed. Accordingly, addressing build and test requirements within this PR does not make much sense

**Comment by philipcardiff** [comment]
> @philipcardiff the build tests are effectively compiling the`feature-petsc-snes-quasi-monolithic` branch, which is still being actively developed. Accordingly, addressing build and test requirements within this PR does not make much sense

True. It will probably only make sense to merge this branch once we have prepared and run all/most of the test cases for the paper (like you did for the higher order).

**Comment by philipcardiff** [comment]
The checks did not all pass, but that's ok since the monolithic branch currently also does not pass, so I have merged it.
